src/Host/Client/Impl/BrokerServices/WebService.cs

```diff
@@ -23,6 +26,19 @@ public class WebService {
                 throw new UnauthorizedAccessException();
             }
 
+            IEnumerable<string> values;
+            if(response.Headers.TryGetValues(CustomHttpHeaders.RTVSApiError, out values)) {
+                var s = values.FirstOrDefault();
+                if (s != null) {
+                    BrokerApiError apiError;
+                    if (Enum.TryParse(s, out apiError)) {
+                        throw new RHostDisconnectedException(apiError);
```

Comment:
This shouldn't be throwing `RHostDisconnectedException`, since this endpoint has nothing to do with host - it's just enumerating sessions. We probably need some kind of generic `BrokerApiException` that wraps `BrokerApiError`, and throw that, or else define custom exception types for every code (but since we can use `catch ... when ...`, I don't see much benefit there). 

`BrokerClient` should then catch that and wrap it into `RHostDisconnectedException` in `CreateBrokerSessionAsync`, like it already does for `HttpRequestException`.